docs/refactoring/confirmation-pages-routing/README.md

```diff
@@ -0,0 +1,78 @@
+- For **/confirm-transaction/** route, the router renders `pages/confirm-transaction` component.
+- For **/confirm-transaction** route `pages/confirm-transaction` component renders `pages/confirm-transaction-switch` by default  (for token methods it renders `pages/confirm-transaction/confirm-token-transaction-switch` which also open `pages/confirm-transaction-switch` by default).
+- `pages/confirm-token-switch` redirect to specific confirmation page route depending on un-approved transaction or signature request in the state.
+- For specific route **/confirm-transaction/${id}/XXXXX** routes also `pages/confirm-transaction` is rendered.
+- Depending on confirmation route `pages/confirm-transaction` and `pages/confirm-transaction/confirm-token-transaction-switch` renders specific confirmation page component.
+
+**Current Route component mapping**
+
+| Route                                           | Component                            |
+| ----------------------------------------------- | ------------------------------------ |
+| /confirm-transaction/${id}/deploy-contract      | pages/confirm-deploy-contract        |
+| /confirm-transaction/${id}/send-ether           | pages/confirm-send-ether             |
+| /confirm-transaction/${id}/send-token           | pages/confirm-send-token             |
+| /confirm-transaction/${id}/approve              | pages/confirm-approve                |
+| /confirm-transaction/${id}/set-approval-for-all | pages/confirm-approve                |
+| /confirm-transaction/${id}/transfer-from        | pages/confirm-token-transaction-base |
+| /confirm-transaction/${id}/safe-transfer-from   | pages/confirm-token-transaction-base |
+| /confirm-transaction/${id}/token-method         | pages/confirm-contract-interaction   |
+| /confirm-transaction/${id}/signature-request    | pages/confirm-signature-request.js   |
+
+## Proposed flow
+
+The proposed routing of confirmation pages looks like.
+
+![Confirmation Pages Routing - Proposed](https://raw.githubusercontent.com/MetaMask/metamask-extension/develop/docs/refactoring/confirmation-pages-routing/proposed.png)
+
+- There are 2 ways in which confirmation pages can be opened:
+  1. User triggers send flow from within Metamask
+     - If the user trigger send flow from within Metamask, user selects receipient and amount and on send screen, at this point an un-approved transaction is created in background and user is re-directed to specific transaction route **/confirm-transaction/${id}/XXXX** depending on transaction.
+  2. DAPP sends request to Metamask
+     - If DAPP send request to Metamask an un-approved transaction or signature request is created in background and UI is triggered to open (if it is not already open).
+     - The router find un-approved transaction in state and re-route to **/confirm-transaction**.
```

Comment:
It was a little tricky for me to understand the changes here. I think I understand now after looking at the charts. 


Maybe we can elaborate a bit more here. 

proposed: 
```suggestion
     - [changed] Instead of rendering `pages/home`,  `pages/routes` finds the unapproved transaction in state and reroutes to **/confirm-transaction**.
```